checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInference.java

```diff
@@ -1,176 +1,873 @@
+   * @param invocation the method invocation node to check
    */
-  private void propagateRegPaths(Block curBlock, Set<Block> visited, Deque<Block> worklist) {
+  private void analyzeOwnershipTransferAtMethodCall(
+      Set<Obligation> obligations, MethodInvocationNode invocation) {
+    List<? extends VariableTree> paramsOfCurrentMethod = methodTree.getParameters();
+    if (paramsOfCurrentMethod.isEmpty()) {
+      return;
+    }
+    List<? extends VariableElement> invocationParams = mcca.getParametersOfInvocation(invocation);
+    if (invocationParams.isEmpty()) {
+      return;
+    }
+    List<Node> arguments = mcca.getArgumentsOfInvocation(invocation);
 
-    List<Block> successors = getNormalSuccessors(curBlock);
+    for (int i = 1; i < arguments.size() + 1; i++) {
+      if (!resourceLeakAtf.hasOwning(invocationParams.get(i - 1))) {
+        continue;
+      }
+      for (int j = 1; j < paramsOfCurrentMethod.size() + 1; j++) {
+        VariableTree paramOfCurrMethod = paramsOfCurrentMethod.get(j - 1);
+        if (resourceLeakAtf.hasEmptyMustCallValue(paramOfCurrMethod)) {
+          continue;
+        }
 
-    for (Block b : successors) {
-      // If b is a special block, it must be the regular exit, since we do not propagate to
-      // exceptional successors.
-      if (b.getType() == Block.BlockType.SPECIAL_BLOCK) {
-        WholeProgramInference wpi = typeFactory.getWholeProgramInference();
-        assert wpi != null : "MustCallInference is running without WPI.";
-        for (VariableElement fieldElt : owningFields) {
-          wpi.addFieldDeclarationAnnotation(fieldElt, OWNING);
+        Node arg = NodeUtils.removeCasts(arguments.get(i - 1));
+        VariableElement paramElt = TreeUtils.elementFromDeclaration(paramOfCurrMethod);
+        if (isParamAndArgAliased(obligations, arg, paramElt)) {
+          addOwningToParam(j);
+          break;
         }
       }
+    }
+  }
 
-      if (visited.add(b)) {
-        worklist.add(b);
+  /**
+   * Checks whether the given argument which is passed in the invocation node and the given
+   * parameter of the current method are resource aliases.
+   *
+   * @param obligations the obligations associated with the current block
+   * @param argument the argument
+   * @param paramElt the parameter
+   * @return true if the {@code paramElt} is in the resource alias set of the given {@code arg}
+   */
+  private boolean isParamAndArgAliased(
+      Set<Obligation> obligations, Node argument, VariableElement paramElt) {
+    Set<ResourceAlias> argAliases = getResourceAliasOfArgument(obligations, argument);
+    for (ResourceAlias argAlias : argAliases) {
+      Element argAliasElt = argAlias.reference.getElement();
+      if (argAliasElt.equals(paramElt)) {
+        return true;
       }
     }
+    return false;
   }
 
   /**
-   * Returns the non-exceptional successors of the current block.
+   * Analyzes the called-methods set of each argument passed into the method call to compute
```

Comment:
This documentation, and the name of the method, starts with low-level operational details.  Instead, make the name and the documentation focus on *what* the method does, not *how* the method does it.  This method computes `@Owning` annotations for arguments of a call.

The same comment applies to many other methods in this class.  (All the ones whose documentation starts with "Analyzes", and others besides.)